### PR TITLE
Country names from CLDR

### PR DESCRIPTION
**Pull request by firstred**
## Description

While checking out CLDR I saw that it contains localized country names as well.
PrestaShop is missing a lot of translations so CLDR also seems to be useful here for country names.
I made a quick 'n dirty Node.js script to generate the translation files: https://gist.github.com/firstred/0b47dac668a520378c788859767a672d
## Steps to Test this Feature

Step 1. Go to Duolingo
Step 2. Complete every course
Step 3. With your newly gained language skills, check all the country names
OR
Step 1. Check out the gist used for the generation of the files: https://gist.github.com/firstred/0b47dac668a520378c788859767a672d
Step 2. Check if the language codes are mapped correctly
Step 3. Check if all files are correct

Whichever you prefer 😜

### CODE REVIEW

**Comment by xBorderie** [comment]
Testing procedure unclear: now أنا أعرف الكونغ فو.

**Comment by julienbourdeau** [comment]
Thank you for this PR !

I believe this has other consequences (like relationship with crowdin) and probably some I don't even know about ^^

**Comment by firstred** [comment]
Let's create another node.js script to analyze the difference? ^^

**Comment by Shudrum** [comment]
Step 1. Go to Duolingo
Step 2. Complete every course
Step 3. With your newly gained language skills, check all the country names

😄😄😄😄😄😄😄😄

File: install-dev/langs/bn/data/country.xml [hunk 1]
```diff
@@ -1,735 +1,777 @@
...
   </country>
-  <country id="MF">
-    <name>Saint Martin</name>
+  <country id="SA">
+    <name>&#2488;&#2508;&#2470;&#2495; &#2438;&#2480;&#2476;</name>
   </country>
-  <country id="PM">
-    <name>Saint Pierre and Miquelon</name>
+  <country id="SB">
+    <name>&#2488;&#2482;&#2507;&#2478;&#2472; &#2470;&#2509;&#2476;&#2496;&#2474;&#2474;&#2497;&#2462;&#2509;&#2460;</name>
```
Comment (mickaelandrieu) [inline install-dev/langs/bn/data/country.xml]: Are you sure this is a good thing ? ^^

Comment (firstred) [inline install-dev/langs/bn/data/country.xml]: `simplexml_load_file` doesn't care.
I wasn't sure if all languages used or might get used in the future by PrestaShop could fit in `UTF-8`. That's the only reason for encoding.

Comment (firstred) [inline install-dev/langs/bn/data/country.xml]: Cool!

Did some research and it all the characters can be represented with `UTF-8`

**Comment by firstred** [comment]
I've removed the xml encoding.

It's XML, so I've just used the raw characters and set encoding to `UTF-8`. That should work well!

**Comment by firstred** [comment]
For future reference, seen the way PrestaShop treats characters (everything = `UTF-8`), the only characters that need to be escaped are:
`<>&'"`

**Comment by vincentbz** [comment]
It's time to take a look on that PR^^

**Comment by kpodemski** [comment]
@vincentbz it's not @firstred fault of course, you should stash them :P

**Comment by xBorderie** [comment]
Well, in any case it cannot be merged as-is: "This branch has conflicts that must be resolved".
Do you think you could rebase it, Michael?

**Comment by mickaelandrieu** [comment]
@vincentbz ``git stash``

**Comment by firstred** [comment]
What's happened here? xD

I'll try to rebase it

**Comment by firstred** [comment]
Done :)

**Comment by xBorderie** [comment]
Thanks @firstred! Could you amend your commit name, please?

**Comment by firstred** [comment]
Pourquoi? :confused:

**Comment by firstred** [comment]
Ohh, it's the improvement sign

**Comment by xBorderie** [comment]
Old habits die hard :)

**Comment by mickaelandrieu** [comment]
Hi everyone,

@AlexEven are you able to check it ? I don't think some of PM or QA team can.

@firstred  What is the risk if we merge it and if we introduce a bug ?

Mickaël

**Comment by firstred** [comment]
I don't see how this can cause a bug. It is suitable for the current
languages, it's data from CLDR and UTF-8. PrestaShop should be capable of
handling it. If it does cause problems, we will probably need to upgrade
PrestaShop (esp. the installer).

**Comment by AlexEven** [comment]
@firstred - I'm only seeing this now, great idea!

I can't see the file for Vietnamese, and I'm not really sure about this, but isn't it also an exception like Slovene or Canadian French? Vi / Vn ?

Thank you!!

and @mickaelandrieu I don't see why PM or QA team wouldn't be able to check: it's making sure the countries are installed in the right languages. QA team should be able to handle it as much as I can :p

**Comment by vincentbz** [comment]
Hi,

Listing is ok :)
image

But I have a DB problem at installation with that PR
image

**Comment by AlexEven** [comment]
@vincentbz You need also to make sure list of countries is correct in Locations > Countries - and  check "tricky" languages (https://gist.github.com/firstred/0b47dac668a520378c788859767a672d#file-index-js-L14-L19)

**Comment by mickaelandrieu** [comment]
@firstred we are testing the installation workflow, some issues
> Notice: Undefined index: an in /home/ubuntu/workspace/PrestaShop/install-dev/classes/LanguageList.php on line 163

Sounds like it's not so easy to accept :(

**Comment by firstred** [comment]
How do I test that? The latest source doesn't even translate the page for me :(

**Comment by firstred** [comment]
Oh, I've got it right now.

@vincentbz I think that you are running a different version. I can only test the public source code.

@mickaelandrieu I'm able to reproduce the issue. I think that the script needs to be run again in order to add the new languages. This PR is quite old.

**Comment by mickaelandrieu** [comment]
@firstred yup sorry, from my experience this kind of huge PR is always hard to be accepted, thank you to stay brave and active we all like you a lot here!

awesome

**Comment by firstred** [comment]
PR updated, but the `an` folder is simply missing.

**Comment by AlexEven** [comment]
@firstred : I think it's not an issue if the `an` folder is missing. It's a long story, but we're using Aragonese as a kind of bridge/neutral language. @djfm would probably better explain it ^^;
In any case we don't support translations in Aragonese, so I'd say we can ignore this.

**Comment by vincentbz** [comment]
@firstred i'm using the develop version^^ and use git checkout countrynames to switch on your branch.
I'll try to see with somebody else

**Comment by firstred** [comment]
@vincentbz Yep, that was not the latest source code
If you `git checkout countrynames` you will use an outdated version. Make sure you `git rebase origin/develop` as well to test the effect of the pull request on the latest source ;)

**Comment by firstred** [comment]
Of course, I'll try to rebase as often as possible, but sometimes PRs can be waiting here for months, making it harder to check these of you only `git checkout pullrequest`.

**Comment by vincentbz** [comment]
Ok i've done a git pull while being on your branch and I don't have the previous errors, thanks :)
Now I have this, during installation 

image

When it's in turkish, it's a database error:
image

**Comment by firstred** [comment]
Running latest source code, in Turkish: http://www.denieuweprestashop.nl/

So I have no idea where those problems come from. It installs just fine.

**Comment by vincentbz** [comment]
Yeah now i'm quite sure it doesn't come from your PR.

**Comment by djfm** [comment]
This looks nice, thanks @firstred.

The diff is a bit hard to analyze because of the size, but as far as I can tell all that this does is change the xml fixtures, so as far as testing is concerned, just checking that the install works in all languages should be close to enough. It's not very hard to write a simple CLI script that performs the CLI install in all languages, or even a selenium test to be on the safe side.

Maybe I've missed it, but I think what would be even more cool is to also commit the script that you used to inject the CLDR data into the fixtures so that next time someone wants to add a language to the installer they can also generate the country names easily.

**Comment by firstred** [comment]
I've pasted everything in a gist: https://gist.github.com/firstred/0b47dac668a520378c788859767a672d
:smile:

**Comment by vTerenti** [comment]
Hi!


Seems to be good for me.

Thanks for your contribution!

Best regards

**Comment by mickaelandrieu** [comment]
@firstred would you mind to rebase ? If tests pass I'll merge :)

**Comment by firstred** [comment]
It still breaks, but I have found out why. The main country xml `data/xml/country.xml` still assumes that countries such as Netherlands Antilles exist, but they don't and can no longer be found in the CLDR data. Gotta find out which countries have been removed/replaced and that should be enough to fix it, I think.

**Comment by firstred** [comment]
And the European Union is being treated as a country by CLDR o.0

**Comment by xBorderie** [comment]
That's weird, it should be a Grouping: http://www.unicode.org/cldr/charts/latest/supplemental/territory_containment_un_m_49.html
Could this be a PS issue rather than a CLDR one?

**Comment by AlexEven** [comment]
:disappointed_relieved:

**Comment by firstred** [comment]
Still interested in this PR? 
I closed it because every time I fixed something, I came across a new mess somehow :/
E.g., removing a country meant adjusting 8293723 another places.

I can have another look if you'd like, but I'm afraid that a large part of the architecture needs to be changed. Not sure if that can still be done in this phase.

**Comment by AlexEven** [comment]
Yes, that what they explained to me. It's just a pity that at this point we can't benefit from your PR. Hopefully in the near future this will change :)
Thanks again for trying!!

**Comment by firstred** [comment]
I see that the CLDR package itself is now included. I think that we will need a totally different approach in 1.7.1. Just extract the data from CLDR directly and skip the XML files. That would remove another dependency and decrease the chance of having outdated information.

**Pull request closed without merging**